### ISSUE TITLE
Accessibility: Improving the web accessibility of Jupyter’s home page

### ISSUE THREAD

**Issue by alexanderchosen** [jupyterhub/outreachy#106]
### What page is this for?

 https://jupyter.org/

### WAVE accessibility report

![wave-jupyterhub1](https://user-images.githubusercontent.com/56475753/197180402-4d995c9b-9ad2-4f72-bd0a-e5f304d6d711.png)

The following are the relevant accessibility report gotten from WAVE:

![wave-jupyterhub](https://user-images.githubusercontent.com/56475753/197180758-3f28b218-83d7-47a0-b716-4b9df326cc4d.png)

1. 7 contrast errors: A Contrast error means that there is very low contrast between the text and background color. For a website to be accessible, it must meet the WCAG contrast requirements [here](https://webaim.org/resources/contrastchecker/#:~:text=The%20Lightness%20slider%20can%20be,such%20as%20form%20input%20borders).).
The contrast error shows that some text with background images in the jupyter’s home page does not provide adequate text contrast and a defined foreground and background color with sufficient contrast which makes it less accessible to all users.

2. 5 alerts: The alert for this page flags images with the same alternative text used, missing h1 heading, and 2 possible headings. Possible headings are used to flag text which appears to be a heading but is not a heading element.

3. 51 features: Features indicate elements that improve accessibility when implemented correctly. The 51 features for this page indicate 20 X alternative text, 30 x linked images with alternative text, and 1 language used. From my careful analysis, some of the elements suggested are not necessary while others will be implemented in the HTML changes written below.

4. 26 structural elements: Structural elements refer to the overall structure/flow of the webpage.
The relevant report from this page is the use of h2 headings in the footer.

### The HTML element to be changed

```html
<!-- CONTRAST ERRORS -->

<!-- The background color used doesn't meet WCAG contrast requirements --> 
  
<a class="orange-button keychainify-checked" href="/try" style="color: rgb(255, 255, 255); background-color: rgb(92, 87, 87);">
Try it in your browser
</a>

<!-- The background color used doesn't meet WCAG contrast requirements-->

<a class="orange-button install-button keychainify-checked" href="/install" style="color: rgb(255, 255, 255); background-color: rgb(92, 87, 87);">
Install JupyterLab
</a>

<!-- The background color used doesn't meet WCAG contrast requirements and doesn't have a defined style attribute--> 

<a class="orange-button keychainify-checked" href="/try">
Try it in your browser
</a>

<!--  The background color used doesn't meet WCAG contrast requirements and doesn't have a defined style attribute --> 

<a class="orange-button install-button keychainify-checked" href="/install">
Install the Notebook
</a>

<!-- The background color used doesn't meet WCAG contrast requirements and doesn't have a defined style attribute -->
 
<a class="orange-button keychainify-checked" href="/hub">
Learn more about JupyterHub
</a>

<!-- The background color used doesn't meet WCAG contrast requirements and doesn't have a defined style attribute --> 

<a class="orange-button keychainify-checked" href="/try">
Try it in your browser
</a>

<!-- The background color used doesn't meet WCAG contrast requirements and doesn't have a defined style attribute-->

<a class="orange-button install-button keychainify-checked" href="/install">
Install Voilà
</a>


<!-- ALERTS -->

<!-- 2 nearby images have the same alternative text  -->

<img class="img-responsive fade-in two" src="/assets/homepage/circle2.svg" width="677" height="286" alt="circle of programming language icons" loading="eager">

<img class="img-responsive fade-in three" src="/assets/homepage/circle3.svg" width="677" height="286" alt="circle of programming language icons" loading="eager">



<!-- FEATURES -->

<!-- Alternative text -->

<!--The length of alternative image text -->

<img src="assets/homepage/labpreview.png" alt="examples of jupyterlab workspaces in single document and multiple document workspaces" width="885" height="627" class="img-responsive" loading="eager">

<!--The length and clear meaning of alternative image text -->

<img src="assets/homepage/jupyterpreview.png" alt="example notebook of Lorenz differential equations" width="885" height="627" class="img-responsive" loading="lazy">


<!--The clear meaning of alternative image text -->

<img src="assets/homepage/kernel.svg" alt="icon that looks like a terminal represents kernel" loading="lazy">


<!--Language -->

<!-- french language not indicated -->

<p class="dropshadow">
Cité des Science, Paris, France
</p>

<!-- Structural elements -->

<!-- H2 element used in footer -->


<h2>
Project Jupyter
</h2>

<h2>
Subprojects
</h2>

<h2>
Follow us
</h2>

```

### Your proposed HTML (or CSS), after change

```html
<!-- Contrast error solutions -->

<!-- The background color's lightness should be reduced thus making the color have a dark shade for more visibility -->

<!-- Also, for uniformity of colors among the buttons, all the background colors should be the same: rgb(144, 63, 20)-->

<a class="orange-button keychainify-checked" href="/try" style="color: rgb(255, 255, 255); background-color: rgb(144, 63, 20);">
Try it in your browser
</a>

<a class="orange-button install-button keychainify-checked" href="/install" style="color: rgb(255, 255, 255); background-color: rgb(144, 63, 20);">
Install JupyterLab
</a>

<a class="orange-button keychainify-checked" href="/try" style="color: rgb(255, 255, 255); background-color: rgb(144, 63, 20);>
Try it in your browser
</a>

<a class="orange-button install-button keychainify-checked" href="/install" style="color: rgb(255, 255, 255); background-color: rgb(144, 63, 20);>
Install the Notebook
</a>

<a class="orange-button keychainify-checked" href="/hub" style="color: rgb(255, 255, 255); background-color: rgb(144, 63, 20);>
Learn more about JupyterHub
</a>

<a class="orange-button keychainify-checked" href="/try" style="color: rgb(255, 255, 255); background-color: rgb(144, 63, 20);>
Try it in your browser
</a>

<a class="orange-button install-button keychainify-checked" href="/install" style="color: rgb(255, 255, 255); background-color: rgb(144, 63, 20);>
Install Voilà
</a>


<!-- Alerts error solution -->
<!--slight changes are made to the image alternative text -->

<img class="img-responsive fade-in two" src="/assets/homepage/circle2.svg" width="677" height="286" alt="circle of programming language icons" loading="eager">

<img class="img-responsive fade-in three" src="/assets/homepage/circle3.svg" width="677" height="286" alt="Another circle of programming language icons" loading="eager">

<!-- alternative image text simplified -->

<img src="assets/homepage/jupyterpreview.png" alt="An example of Lorenz differential equations notebook" width="885" height="627" class="img-responsive" loading="lazy">

<img src="assets/homepage/kernel.svg" alt="icon resembling a terminal represents kernel" loading="lazy">


<!--Language -->
<!-- french language properly indicated -->

<p lang="fr" class="dropshadow">
Cité des Science, Paris, France
</p>


<!-- Structural element -->
<!-- h2 element changed to h5 in footer: The h2 element tag doesn't fit the desired font outcome  -->

<h5>
Project Jupyter
</h5>

<h5>
Subprojects
</h5>

<h5>
Follow us
</h5>

```

### MY SUGGESTIONS 

From my careful analysis of Jupyter's home page, I suggest the following changes:
1. The responsiveness of the div image containing the Jupytercon logo, date, and title: The Image containing the jupytercon logo and other information at the top of the home page is not responsive.

![jupyter-responsive](https://user-images.githubusercontent.com/56475753/197174889-371b67ee-3466-44c6-980d-1d77fef02847.png)

I carefully observed that the div which was styled using Bootstrap and the jupytercon logo is not responsive on a small device screen. This is an accessibility issue as users with small screens will be affected.

2. The contrast error suggests that the light intensity of the background color for all the buttons should be reduced thus giving us a dark variant of the initial primary color used. This is in order to meet the Web content accessibility guidelines (WCAG). 

![wave-color-1](https://user-images.githubusercontent.com/56475753/197177595-a5413e4b-7138-4eb3-8762-8dc1ec7f64f9.png)

From the image shown above, the color (#903F14 or rgb(144,63,20)) gotten gives the webpage a different and too colorful look. The color doesn't blend well with other colors and needs a change.

I suggest the use of other colors that blends with the primary color of the home page. In order to give the home page a good color match, I decided to try out a dark variant of the icon color (hex - #5c5757 or rgb(92,87,87)).

![wave-color-2](https://user-images.githubusercontent.com/56475753/197177671-01756d60-a840-4c9f-a465-f060bc489630.png)

The new color rhymes and blends in with other colors used better than the other variant gotten from the initial color.

**Comment by alexanderchosen** [comment]
Hello mentors @sgibson91 @minrk @GeorgianaElena, I have carefully worked on improving the web accessibility of Jupyter's home page [here](https://jupyter.org/). Please help review the issues submitted and I hope to get your feedback and corrections.
Thanks.

**Comment by minrk** [comment]
Thank you for the suggestions and detailed explanations, and trying different colors!

**Comment by alexanderchosen** [comment]
@minrk thanks for your quick feedback. I'm so glad for this opportunity to contribute to this amazing project.
Thanks.

**Comment by sgibson91** [comment]
Thank you for your contribution! Since the contribution period is now over, we will close this issue.
